File: AlphaStream/Models/BidResult.py
from datetime import datetime
from .Bid import Bid

class BidResult(object):
    """Result of a bid to license an Alpha."""

    def __init__(self, json):

        # True if the bid is successful
        self.Success = json.get('success', None)

        # Alpha's Insight magnitude accuracy
        self.ActiveBid = json.get('active-bid', None)
        if self.ActiveBid:
            self.ActiveBid = Bid(self.ActiveBid)

        # Time that the next auction will occur
        self.NextAuctionTime = datetime.utcfromtimestamp(json.get('next-auction-time', 0))

        # Alpha's capacity for the next auction
        self.NextAuctionCapacity = json.get('next-auction-capacity', 0)

        # Minimum capital required to place a bid
        self.MinimumCapitalRequired = json.get('minimum-capital-required', 0)

    def __repr__(self):
        if not self.Success:
            return 'Unsuccessful bid'
        return f'''{self.ActiveBid}
    Next auction time: {self.NextAuctionTime}
    Next auction capacity: {self.NextAuctionCapacity}
    Minimum capital required: {self.MinimumCapitalRequired}'''

File: QuantConnect.AlphaStream/Models/BidResult.cs
using System;
using Newtonsoft.Json;
using QuantConnect.Util;

namespace QuantConnect.AlphaStream.Models
{
    /// <summary>
    /// Result of a bid to license an Alpha
    /// </summary>
    public class BidResult
    {
        /// <summary>
        /// True if the bid is successful
        /// </summary>
        [JsonProperty("success")]
        public bool Success { get; set; }

        /// <summary>
        /// Information about the active bid
        /// </summary>
        [JsonProperty("active-bid")]
        public Bid ActiveBid { get; set; }

        /// <summary>
        /// Time that the next auction will occur
        /// </summary>
        [JsonProperty("next-auction-time"), JsonConverter(typeof(DoubleUnixSecondsDateTimeJsonConverter))]
        public DateTime NextAuctionTime { get; set; }

        /// <summary>
        /// Alpha's capacity for the next auction
        /// </summary>
        [JsonProperty("next-auction-capacity")]
        public decimal NextAuctionCapacity { get; set; }

        /// <summary>
        /// Minimum capital required to place a bid
        /// </summary>
        [JsonProperty("minimum-capital-required")]
        public decimal MinimumCapitalRequired { get; set; }
    }
}